lib/IAMClient.js

```diff
@@ -395,13 +400,33 @@ class IAMClient {
                 ret += receivedData.toString();
             })
 
-            .on('error', callback)
+            .on('error', err => {
+                if (logger) {
+                    logger.debug('error receiving data', {
+                        component: 'vaultclient',
+                        method: 'IAMClient:request()',
+                        error: err.stack || err,
+                    });
+                    return callback(errors.InternalError);
```

Comment:
I do not get the point here. Why do you change the error when there is a logger :s